cmd/cdi-importer/importer.go

```diff
@@ -271,34 +272,45 @@ func newDataSource(source string, contentType string, volumeMode v1.PersistentVo
 	case cc.SourceHTTP:
 		ds, err := importer.NewHTTPDataSource(getHTTPEp(ep), acc, sec, certDir, cdiv1.DataVolumeContentType(contentType))
 		if err != nil {
-			errorCannotConnectDataSource(err, "http")
+			errorCannotConnectDataSource(err, cc.SourceHTTP)
 		}
 		return ds
 	case cc.SourceImageio:
 		ds, err := importer.NewImageioDataSource(ep, acc, sec, certDir, diskID, currentCheckpoint, previousCheckpoint)
 		if err != nil {
-			errorCannotConnectDataSource(err, "imageio")
+			errorCannotConnectDataSource(err, cc.SourceImageio)
 		}
 		return ds
 	case cc.SourceRegistry:
 		ds := importer.NewRegistryDataSource(ep, acc, sec, certDir, insecureTLS)
 		return ds
 	case cc.SourceS3:
-		ds, err := importer.NewS3DataSource(ep, acc, sec, certDir)
+		var (
+			ds  *importer.S3DataSource
+			err error
+		)
+		if serviceAccount != "" {
```

Comment:
Is the value of `serviceAccount` ever used in the aws API? Or does the aws lib assume that the pod is running as the privileged service account?

Reply:
The latter! hopefully i was too far off base, but the approach was that this would be used as a "flag" given the service account should have been properly set up by the time it is invoked here? (or fail on referencing an un-instantiated SA)